### PR TITLE
Rename location to project

### PR DESCRIPTION
**Pull request by anikaweinmann**
- Add project endpoints
- Add G83/G84 tests
- rename location to project

### CODE REVIEW

File: src/actinia_module_plugin/endpoints.py [hunk 1]
```diff
@@ -45,6 +45,36 @@
 from actinia_module_plugin.api.actinia_templates import ActiniaTemplate
 from actinia_module_plugin.api.actinia_templates import ActiniaTemplateId
 
+from actinia_core.endpoints import get_endpoint_class_name
```
Comment (mmacata) [inline src/actinia_module_plugin/endpoints.py:48]: Isn't this a circular reference? Does ist start like this?

File: src/actinia_module_plugin/endpoints.py [hunk 2]
```diff
@@ -45,6 +45,36 @@
...
+
+
+def create_project_endpoints(apidoc, projects_url_part="projects"):
+    """
+    Function to add resources with "project" inside the endpoint url.
```
Comment (mmacata) [inline src/actinia_module_plugin/endpoints.py:53]:
```suggestion
    Function to add resources with "projects" inside the endpoint url.
```

File: src/actinia_module_plugin/endpoints.py [hunk 3]
```diff
@@ -76,15 +106,11 @@ def create_endpoints(flask_api):
     apidoc.add_resource(ListVirtualModules, "/modules")
     apidoc.add_resource(DescribeVirtualModule, "/modules/<module>")
 
-    apidoc.add_resource(
-        GdiAsyncEphemeralExportResource,
-        "/locations/<string:location_name>/processing_export",
-    )
+    # add deprecated location endpoints
+    create_project_endpoints(apidoc, projects_url_part="locations")
```
Comment (mmacata) [inline src/actinia_module_plugin/endpoints.py:111]:
```suggestion
```
(Suggestions are a bit tricky here - these lines are added below)

File: src/actinia_module_plugin/endpoints.py [hunk 4]
```diff
@@ -76,15 +106,11 @@ def create_endpoints(flask_api):
...
 
-    apidoc.add_resource(
-        GdiAsyncPersistentResource,
-        "/locations/<string:location_name>/mapsets/<string:mapset_name>/processing",
-    )
+    # add project endpoints
+    create_project_endpoints(apidoc, projects_url_part="projects")
```
Comment (mmacata) [inline src/actinia_module_plugin/endpoints.py:113]:
```suggestion
    # Endpoints for processing including deprecated locations endpoints
    create_project_endpoints()
    create_project_endpoints(projects_url_part="locations")
```
Suggestion is to align it to actinia-core.
What happens with `apidoc`? Does this need to be adjusted in actinia-core?

File: tests/testsuite.py [hunk 5]
```diff
@@ -76,6 +76,14 @@ class ActiniaTestCase(unittest.TestCase):
     user = None
     auth_header = {}
     users_list = []
+    project_url_part = "projects"
```
Comment (mmacata) [inline tests/testsuite.py:79]: For the plugin, `project_url_part` were imported from actinia-core (if no circular reference).
Is this also possible here?

**Review by mmacata — APPROVED** [review]
Thank you!
In general I approve, only some minor comments.

File: src/actinia_module_plugin/endpoints.py (hunk 1 above)
Comment (anikaweinmann) [reply src/actinia_module_plugin/endpoints.py:48]: Yes it started.

Comment (anikaweinmann) [reply src/actinia_module_plugin/endpoints.py:48]: Yes it did

File: src/actinia_module_plugin/endpoints.py (hunk 4 above)
Comment (anikaweinmann) [reply src/actinia_module_plugin/endpoints.py:113]: We discussed it and I will make it more like actinia-core.

**Pull request merged**